src/main/java/org/andersen/homework/model/entity/user/User.java

```diff
@@ -1,20 +1,35 @@
 package org.andersen.homework.model.entity.user;
 
+import jakarta.persistence.DiscriminatorColumn;
+import jakarta.persistence.DiscriminatorType;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Inheritance;
+import jakarta.persistence.InheritanceType;
+import jakarta.persistence.Table;
 import java.util.UUID;
 import lombok.AllArgsConstructor;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
 import lombok.ToString;
-import org.andersen.homework.model.enums.UserRole;
 
+@Entity
+@Table(name = "\"user\"")
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
```

Comment:
Why have you chosen this InheritanceType?

Reply:
It's a simple way to store parents and children without creating extra tables.